### ISSUE TITLE
Test to ensure `selected` attribute exists on a select/option dropdown

### ISSUE THREAD

**Issue by SYU15** [ember-a11y/ember-a11y-testing#172]
I think it may be a good idea to add this test. If the `selected` attribute is missing in the list of options, then users with screen readers (JAWS and NVDA specifically) cannot understand/hear which item is being chosen.

Example:
```
<label for="answer">Choose an answer:</label>

<select name="answer" id="answer">
  <option selected value="yes">Yes</option>
  <option value="no">No</option>
  <option value="maybe">Maybe</option>
</select>
```

**Comment by SYU15** [comment]
Will explore `aria-selected` as a solution to gaps in screen reader technology.

**Comment by MelSumner** [comment]
Note: we need to review HTML spec to see if the `selected` attribute maps to the `selected` property or the `defaultSelected` property.

Here's what should happen for a `<select>` element:

1. Using VoiceOver with Safari for Mac: 
* on focus, it says the name of the selected option (if it's not blank) then the element's label. 
* on navigating the option list, each option's name should be announced 
* on selected, using VO + `SPACE`, it should announce "press [option name]." 
* if option is selected using the Return/Enter key instead, it does not say "press [option name]." 
* regardless of how the selection is made, focus should return to the `<select>` and it again says the name of the selected option and the element's label.

2. Using NVDA with Firefox on Windows:
* on focus, it says the label + "combobox" + currently selected option + "collapsed" 
* on `ARROW` up or down (without expanding the dropdown), it says the name of the option that is brought into focus
* on `SPACE` it says "combobox expanded" + currently selected option + "list" + position of option in the list 
* navigating through the options with the arrow keys will cause the selected option's name to be announced as well as its position in the list (such as "dog, 3 of 3") 
* pressing the `ENTER` key will return focus to the `<select>` and read out the label + "combobox" + option selected + "collapsed"

**Comment by MelSumner** [comment]
also + @rwjblue and @wycats since there was some question about what NVDA + Firefox does with a native select element.

**Comment by drewlee** [comment]
@SYU15, this project doesn't actually handle the logic behind the specific rules/checks that carry out the a11y assessment. That is handled by [axe-core](https://github.com/dequelabs/axe-core/) and should be filed there. IMO, this sounds like an issue with the specific vendors (JAWS and NVDA), since they chose not to adhere to the HTML specification. The `selected` attribute is optional and is only necessary to define the initial selected `option`. Applying it after a user made a selection seems redundant and hacky. The `aria-selected` property should only be used if we're authoring a custom combobox that mimics the behavior of an HTML `select` element. It would likely be better to file this as a bug with JAWS and NVDA, as an author omitting a `selected` attribute is inherently doing nothing wrong.

**Comment by drewlee** [comment]
I'd like to close this issue out as it would need to be filed against [axe-core](https://github.com/dequelabs/axe-core).

**Comment by SYU15** [comment]
Sounds good @drewlee @MelSumner thanks for the explanation. In the meantime, we will add both `selected` and `aria-selected` to our option elements to ensure a range of screen readers will be able to read out the selected option.
